lightkurve/seismology/core.py

```diff
@@ -280,11 +354,175 @@ def plot_echelle(self, deltanu=None, numax=None,
+                                           scale=scale)
+
+        fig = figure(plot_width=plot_width, plot_height=plot_height,
+                     x_range=(0, 1), y_range=(y_f[0].value, y_f[-1].value),
+                     title=title, tools='pan,box_zoom,reset',
+                     toolbar_location="above",
+                     border_fill_color="white")
+
+        fig.yaxis.axis_label = r'Frequency [{}]'.format(freq.unit.to_string())
+        fig.xaxis.axis_label = r'Frequency / {:.3f} Mod. 1'.format(deltanu)
+
+        lo, hi = np.nanpercentile(ep.value, [0.1, 99.9])
+        vlo, vhi = 0.3 * lo, 1.7 * hi
+        vstep = (lo - hi)/500
+        color_mapper = LogColorMapper(palette="RdYlGn10", low=lo, high=hi)
+
+        fig.image(image=[ep.value], x=0, y=y_f[0].value,
+                  dw=1, dh=y_f[-1].value,
+                  color_mapper=color_mapper, name='img')
+
+        stretch_slider = RangeSlider(start=vlo,
+                                     end=vhi,
+                                     step=vstep,
+                                     title='',
+                                     value=(lo, hi),
+                                     orientation='vertical',
+                                     width=10,
+                                     height=230,
+                                     direction='rtl',
+                                     show_value=False,
+                                     sizing_mode='fixed',
+                                     name='stretch')
+
+        def stretch_change_callback(attr, old, new):
+            """TPF stretch slider callback."""
+            fig.select('img')[0].glyph.color_mapper.high = new[1]
+            fig.select('img')[0].glyph.color_mapper.low = new[0]
+
+        stretch_slider.on_change('value', stretch_change_callback)
+        return fig, stretch_slider
+
+
+    def interact_echelle(self, notebook_url="localhost:8888", **kwargs):
+        """Display an interactive Jupyter notebook widget showing an Echelle diagram.
+
+        This feature only works inside an active Jupyter Notebook, and
+        requires an optional dependency, ``bokeh`` (v1.0 or later).
+        This dependency can be installed using e.g. `conda install bokeh`.
+
+        Parameters
+        ----------
+        notebook_url : str
+            Location of the Jupyter notebook page (default: "localhost:8888")
+            When showing Bokeh applications, the Bokeh server must be
+            explicitly configured to allow connections originating from
+            different URLs. This parameter defaults to the standard notebook
+            host and port. If you are running on a different location, you
+            will need to supply this value for the application to display
+            properly. If no protocol is supplied in the URL, e.g. if it is
+            of the form "localhost:8888", then "http" will be used.
+        """
+        try:
+            import bokeh
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [redefinition of unused 'bokeh' from line 21](https://app.codacy.com/manual/barentsen/lightkurve/pullRequest?prid=4468966)